source/disruption/disruption.h

```diff
@@ -36,44 +36,51 @@ www.navitia.io
 
 namespace navitia { namespace disruption {
 
+struct Comp{
+    bool operator()(const boost::shared_ptr<type::new_disruption::Impact>& lhs,
+                    const boost::shared_ptr<type::new_disruption::Impact>& rhs);
+};
+
+using DisruptionSet = std::set<boost::shared_ptr<type::new_disruption::Impact>, Comp>;
 
-struct disrupt{
+struct Disrupt{
     type::idx_t idx;
-    type::idx_t network_idx;
-    std::vector<type::idx_t> line_idx;
-    std::vector<type::idx_t> stop_area_idx;
+    const type::Network* network = nullptr;
+    DisruptionSet network_disruptions;
+    std::vector<std::pair<const type::Line*, DisruptionSet>> lines;
+    std::vector<std::pair<const type::StopArea*, DisruptionSet>> stop_areas;
```

Comment:
I don't really need the additional functionalities, and I loose the ability to create them inline